### ISSUE TITLE
Errors thrown by callback swallowed

### ISSUE THREAD

**Issue by zbjornson** [feross/queue-microtask#2]
Hey there,

Errors thrown by the queued callback will get swallowed, there should be a `catch` on the promise. @domenic corrected this in the polyfill I'd posted on MDN (which was removed for some reason; I just restored it). https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/queueMicrotask#Polyfill

**Comment by feross** [comment]
@zbjornson Thanks for the issue. This is the code I found on MDN:

```js
if (typeof window.queueMicrotask !== "function") {
  window.queueMicrotask = function (callback) {
    Promise.resolve()
      .then(callback)
      .catch(e => setTimeout(() => { throw e; }));
  };
}
```

Can you explain why the exception is thrown in a `setTimeout`? Is that so that it can't be caught by another `.catch` at a higher level?

**Comment by zbjornson** [comment]
I think it's this difference, from the OP of https://github.com/whatwg/html/issues/512:

> `Promise.resolve().then(f)` ... creates a promise object and stores any thrown errors there, instead of letting them propagate to "report an exception"

https://html.spec.whatwg.org/multipage/webappapis.html#report-the-error

With `setTimeout`, the error will invoke `window.onerror` (like `queueMicrotask`). Without `setTimeout`, it won't.

```js
> window.onerror = e => console.log("onerror:", e);

> queueMicrotask(() => { throw new Error("eek"); })
"onerror: Script error."
❌Uncaught Error: eek

> Promise.resolve().then(() => { throw new Error("oops"); }).catch(e => setTimeout(() => { throw e; }))
"onerror: Script error."
❌Uncaught Error: oops

> Promise.resolve().then(() => { throw new Error("oops"); }).catch(e => { throw e; })
// no onerror handler
❌Uncaught (in promise) Error: oops
```

**Comment by feross** [comment]
Thanks for the explanation. I released a fix for this as 1.1.1
